python/datafusion/substrait.py

```diff
@@ -15,9 +15,156 @@
 # specific language governing permissions and limitations
 # under the License.
 
+from __future__ import annotations
 
-from ._internal import substrait
+from ._internal import substrait as substrait_internal
 
+from typing import TYPE_CHECKING
 
-def __getattr__(name):
-    return getattr(substrait, name)
+if TYPE_CHECKING:
+    from datafusion.context import SessionContext
+    from datafusion._internal import LogicalPlan
+
+
+class plan:
+    def __init__(self, plan: substrait_internal.plan) -> None:
+        self.plan_internal = plan
+
+    def encode(self) -> bytes:
+        """Encode the plan to bytes.
+
+        Returns
+        -------
+        bytes
+            Encoded plan.
+        """
+        return self.plan_internal.encode()
+
+
+class serde:
+    @staticmethod
+    def serialize(sql: str, ctx: SessionContext, path: str) -> None:
```

Comment:
Would this work on `pathlib.Path`?

Reply:
Updated to accept path. Not yet resolving this comment until unit test is added.

Reply:
Unit tests added